app/src/main/kotlin/com/simplemobiletools/flashlight/activities/BrightDisplayActivity.kt

```diff
@@ -1,66 +1,112 @@
+import com.simplemobiletools.flashlight.helpers.SleepTimer
 import com.simplemobiletools.flashlight.helpers.stopSleepTimerCountDown
-import com.simplemobiletools.flashlight.models.Events
-import org.greenrobot.eventbus.EventBus
-import org.greenrobot.eventbus.Subscribe
-import org.greenrobot.eventbus.ThreadMode
+import com.simplemobiletools.flashlight.screens.BrightDisplayScreen
+import com.simplemobiletools.flashlight.views.AnimatedSleepTimer
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import kotlin.system.exitProcess
 
-class BrightDisplayActivity : SimpleActivity() {
-    private val binding by viewBinding(ActivityBrightDisplayBinding::inflate)
+class BrightDisplayActivity : ComponentActivity() {
+    private val viewModel by viewModels<BrightDisplayViewModel>()
+    private val preferences by lazy { config }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        window.addFlags(
-            WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD or
-                WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED or
-                WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON or
-                WindowManager.LayoutParams.FLAG_FULLSCREEN
-        )
+        if (isOreoMr1Plus()) {
+            setShowWhenLocked(true)
+            setTurnScreenOn(true)
+        } else {
+            window.addFlags(
+                WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD or
+                    WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED or
+                    WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON or
+                    WindowManager.LayoutParams.FLAG_FULLSCREEN
+            )
+        }
 
-        useDynamicTheme = false
         super.onCreate(savedInstanceState)
-        setContentView(binding.root)
-        supportActionBar?.hide()
-        setBackgroundColor(config.brightDisplayColor)
-
-        binding.brightDisplayChangeColor.setOnClickListener {
-            ColorPickerDialog(this, config.brightDisplayColor, true, currentColorCallback = {
-                setBackgroundColor(it)
-            }) { wasPositivePressed, color ->
-                if (wasPositivePressed) {
-                    config.brightDisplayColor = color
-
-                    val contrastColor = color.getContrastColor()
-                    binding.brightDisplayChangeColor.apply {
-                        setTextColor(contrastColor)
-                        background.applyColorFilter(contrastColor)
+        enableEdgeToEdgeSimple()
+        setContent {
+            AppThemeSurface {
+                val colorPickerDialogState = getColorPickerDialogState()
+
+                ScreenContent(colorPickerDialogState::show)
+            }
+        }
+
+        if (isOreoMr1Plus()) {
+            WindowCompat.getInsetsController(window, window.decorView.rootView)
+        }
```

Comment:
do we even need this code?

Reply:
Wow, this does nothing :smile: 
Not sure why I left it like this, I wanted this to be a replacement for that `FLAG_FULLSCREEN` part of these legacy flags, but then again, we have `enableEdgeToEdgeSimple()` here, so I don't think this is needed, even if it was written correctly